src/api.py

```diff
@@ -0,0 +1,44 @@
+import hashlib
+import requests
+
+class Response:
+    def __init__(self, filename, type_of_data, size, url, checksum):
+        self.filename = filename
+        self.type_of_data = type_of_data
+        self.size=size
+        self.url=url
+        self.checksum=checksum
+
+
+def generate_filename(url):
+    """
+    split url on basis of '/' and start iterating from end till you find non empty value
+    """
+    for i in range(len(url.split('/'))-1):
+        name = url.split('/')[-(i+1)]
+        if not(name =='' ):
+            break
+    return name
+
+
+def fetch(url,location):
+    # Fetching data from URL
+    try:
+        con = requests.head(url)
+        r = requests.get(url)
+    except Exception as e:
+        print('Can not get response from url')
+        print(e)
+    # Getting filename
+    name = generate_filename(url = url)
+    filename = location + name
+    # Writing file in the given location
+    try:
+        with open(filename,'wb') as f:
+            f.write(r.content)
+    except Exception as e:
+        print('Can not write on that location')
+        print(e)
+    # Generating response object
+    resp = Response(filename = filename, type_of_data = con.headers['content-type'], size = con.headers['content-length'], url = url, checksum = 0)
```

Comment:
Please format this with each arg on its own line. Also in PEP8 we do not use spaces around equals when used in keyword arguments of a function call.